python/cuml/internals/array.py

```diff
@@ -1104,6 +1104,11 @@ def from_input(
             X = X.to_cupy(copy=False)
         elif isinstance(X, (PandasDataFrame, PandasSeries)):
             X = X.to_numpy(copy=False)
+            # by default pandas converts to numpy 'C' major, which
+            # does not keep the original order
+            if order == "K":
+                X = X.reshape(X.shape, order="F")
```

Comment:
I didn't know that you could change the order via `reshape` and was wondering whether or not this leads to a copy of the data. So I played around a bit to see it happen with my own eyes.
```python
df = pd.DataFrame({'a': [1,2,3], 'b':[11,22,33]})
X = df.to_numpy()
print(X.flags) # C_CONTIGUOUS is false, F_CONTIGUOUS is true
X = X.reshape(X.shape, order="F")
print(X.flags) # C_CONTIGUOUS is false, F_CONTIGUOUS is true :-/
```
Which left my puzzled. Is my assumption of what should be happening wrong? There is `np.asfortranarray()` which does change C to F, but at the cost of making a copy of the data.

Reply:
Ok, got myself confused. For me the result of `to_numpy()` is already F contiguous?!

Reply:
seems like the result can be inconsistent depending on the wrapping that cudf is doing

Reply:
That said, it should have no cost if it's already in the correct order